crates/terminal/src/terminal.rs

```diff
@@ -595,7 +595,8 @@ pub struct Terminal {
 pub struct TaskState {
     pub id: TaskId,
     pub label: String,
-    pub completed: bool,
+    pub running: bool,
+    pub completed_successfully: Option<bool>,
```

Comment:
Could we consolidate these 2 members into a single enum? something like:
```
enum TaskCompletionStatus {
  Cancelled,
  Running,
  Completed { success: bool }
}
```

Reply:
Sure, added this with a few small tweaks around.